app/models/user/permissions.rb

```diff
@@ -60,11 +59,11 @@ def member_permission_ids=(roles)
   end
 
   def member_permission_service_ids=(service_ids)
-    if service_ids.present?
-      service_ids = Array(service_ids).compact.map(&:to_i)
+    if service_ids.is_a? Array
+      service_ids = (service_ids.compact_blank - [EMPTY_VALUE]).map(&:to_i)
```

Comment:
Somebody not very smart decided to use `"[]"` as the value for setting an empty array [in this PR](https://github.com/3scale/porta/pull/121/files#diff-1da47353f13328f81e6ba8800cd866fa4aecbb1d5c5938bd1ece6b2c1e1289c3R3982)

This resulted in array `["[]"]` here, which when converted to integers ended up being `[0]`. The outcome was fine, because `existing_service_ids` never contains 0. But I still think it's not cool, because theoretically maybe there might be service ID that equals `0`.


On the other hand, maybe we shouldn't treat `"[]"` specifically, and discard any other character that is not integer-like, because the user can set anything like `"xyz"`, and it will also be converted to `0`.